protocols/relay/examples/relay.rs

```diff
@@ -173,3 +232,47 @@ impl From<()> for Event {
         Event::Relay(())
     }
 }
+
+#[derive(Debug, StructOpt)]
+enum Mode {
+    Relay,
+    ClientListen,
+    ClientDial,
+}
+
+impl FromStr for Mode {
+    type Err = ModeError;
+    fn from_str(mode: &str) -> Result<Self, Self::Err> {
+        match mode {
+            "relay" => Ok(Mode::Relay),
+            "client-listen" => Ok(Mode::ClientListen),
+            "client-dial" => Ok(Mode::ClientDial),
+            _ => Err(ModeError {}),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct ModeError {}
+impl Error for ModeError {}
+impl fmt::Display for ModeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Could not parse a mode")
+    }
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "libp2p relay")]
+struct Opt {
+    /// The mode (relay, client-listen, client-dial)
+    #[structopt(long)]
+    mode: Mode,
+
+    /// Fixed value to generate deterministic peer id
+    #[structopt(short = "d")]
```

Comment:
```suggestion
    #[structopt(long)]
```

I would suggest using the long command line flag here to avoid confusion. I am not sure how `d` would remind people of `secret_key_seed`.

Reply:
Fixed